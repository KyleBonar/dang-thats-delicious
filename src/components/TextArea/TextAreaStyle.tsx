import styled from "styled-components";

export const StyledDiv = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;

  p {
    font-size: 0.85rem;
    margin-top: 1px;
    margin-bottom: 15px;
  }
`;

export const StyledTextArea = styled.textarea`
  width: 100%;
  max-width: 100%;
  padding: 10px;
  box-sizing: border-box;
  border: 1px solid #ababab;
  margin-top: 5px;
  margin-bottom: 15px;
  min-width: 100%;
  min-height: 35px;
  max-height: 220px;
`;
StyledTextArea.displayName = "StyledTextArea";
