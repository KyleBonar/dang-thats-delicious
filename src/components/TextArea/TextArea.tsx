import * as React from "react";

import { StyledDiv, StyledTextArea } from "./TextAreaStyle";
import Label from "../Label/Label";

export interface TextAreaProps {
  id: string;
  name?: string;
  placeholder?: string;
  required?: boolean;
  label?: string;
  showLabel?: boolean;
  value?: string | number;
  className?: string;
  disabled?: boolean;
  readOnly?: boolean;
  maxLength?: number;
  requirementText?: string;
  onChange: (
    event:
      | React.ChangeEvent<HTMLInputElement>
      | React.ChangeEvent<HTMLTextAreaElement>
  ) => void;
}

const TextArea: React.FunctionComponent<TextAreaProps> = props => {
  const {
    showLabel = false,
    required = false,
    disabled = false,
    readOnly = false,
    id,
    className,
    label,
    placeholder,
    onChange,
    value,
    maxLength,
    requirementText,
    name
  } = props;

  return (
    <StyledDiv className={className}>
      {showLabel && label && (
        <Label htmlFor={id}>
          {label}
          {required ? <span style={{ color: "#B20000" }}>*</span> : ""}
        </Label>
      )}
      <StyledTextArea
        id={id}
        name={name}
        cols={30}
        rows={10}
        placeholder={placeholder}
        onChange={onChange}
        value={value}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        maxLength={maxLength}
        style={{
          marginBottom: requirementText && "0px"
        }}
      />
      {requirementText ? <p>{requirementText}</p> : ""}
    </StyledDiv>
  );
};
export default TextArea;
