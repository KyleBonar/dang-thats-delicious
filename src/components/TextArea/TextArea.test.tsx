// import "jsdom-global/register";
import * as React from "react";
import * as enzyme from "enzyme";

import TextArea from "./TextArea";

let error = console.error;

describe("<TextArea />", () => {
  it("renders", () => {
    const wrapper = enzyme.render(<TextArea />);

    expect(wrapper).toBeTruthy();
  });
});
