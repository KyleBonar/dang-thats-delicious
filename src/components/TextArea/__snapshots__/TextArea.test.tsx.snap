// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> matches snapshot 1`] = `
<div
  className="modi reprehenderit omnis adipisci eius laborum eius"
>
  <textarea
    cols={30}
    disabled={true}
    id="sequi facilis corporis sed veritatis quidem praesentium"
    name="magnam fugiat error molestiae laborum est possimus"
    onChange={[MockFunction]}
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
  />
  <p>
    delectus vel ipsa tempora aut sint aperiam
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 2`] = `
<div
  className="cum ducimus nemo doloremque maiores esse atque"
>
  <textarea
    cols={30}
    disabled={false}
    id="occaecati illum animi vel provident dicta perferendis"
    maxLength={1700}
    onChange={[MockFunction]}
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="aliquam sed sapiente soluta corporis commodi eaque"
  />
  <p>
    itaque voluptas ea ducimus autem sit aspernatur
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 3`] = `
<div>
  <textarea
    cols={30}
    disabled={false}
    id="aspernatur quam eligendi est fugiat dolorem nihil"
    maxLength={4300}
    name="quam nulla est unde sit ex expedita"
    onChange={[MockFunction]}
    placeholder="doloremque explicabo eaque necessitatibus sit suscipit omnis"
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
  />
  <p>
    eveniet assumenda quis quae iusto sapiente nostrum
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 4`] = `
<div
  className="ut occaecati at odit tempora at modi"
>
  <textarea
    cols={30}
    disabled={false}
    id="aut corrupti sed consectetur soluta inventore laudantium"
    onChange={[MockFunction]}
    placeholder="cumque nihil iste est unde rem similique"
    readOnly={true}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="quam qui facere in ut odit animi"
  />
  <p>
    saepe assumenda deserunt consequatur itaque et in
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 5`] = `
<div
  className="possimus earum animi quis expedita et optio"
>
  <textarea
    cols={30}
    disabled={false}
    id="fugit in rerum libero odit aut labore"
    onChange={[MockFunction]}
    placeholder="ducimus fugiat et et aut quidem voluptatibus"
    readOnly={false}
    required={true}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="quos minus deserunt nihil quasi minus quo"
  />
  <p>
    sunt nesciunt provident eaque atque quos rerum
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 6`] = `
<div
  className="a culpa facilis eligendi voluptatibus similique blanditiis"
>
  <textarea
    cols={30}
    disabled={true}
    id="nobis nam dolor quas repellendus in dolores"
    onChange={[MockFunction]}
    placeholder="molestiae cupiditate autem consequatur consequatur accusantium molestiae"
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="non quis quis nobis repudiandae aliquam molestiae"
  />
  <p>
    est eius dolorem amet quia quia aut
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 7`] = `
<div
  className="accusantium vel enim sint voluptatem porro ut"
>
  <textarea
    cols={30}
    disabled={true}
    id="sint cumque sunt mollitia quibusdam officia eum"
    onChange={[MockFunction]}
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
  />
  <p>
    aut distinctio molestias autem expedita omnis sit
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 8`] = `
<div>
  <Label
    htmlFor="vitae consequuntur non sit et ut voluptatem"
  >
    dicta odit vel quisquam qui harum a
    <span
      style={
        Object {
          "color": "#B20000",
        }
      }
    >
      *
    </span>
  </Label>
  <textarea
    cols={30}
    disabled={false}
    id="vitae consequuntur non sit et ut voluptatem"
    onChange={[MockFunction]}
    readOnly={true}
    required={true}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="ea voluptatem totam occaecati eveniet aut eligendi"
  />
  <p>
    ratione eveniet ea totam aut aut sunt
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 9`] = `
<div>
  <textarea
    cols={30}
    disabled={false}
    id="enim rem qui hic dolore quibusdam quasi"
    name="quisquam ut libero explicabo at quae autem"
    onChange={[MockFunction]}
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="modi molestiae rerum non quis sed ea"
  />
  <p>
    repudiandae possimus dignissimos esse qui est aut
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 10`] = `
<div>
  <textarea
    cols={30}
    disabled={false}
    id="laborum nesciunt dolor vero voluptate esse aliquam"
    maxLength={2706}
    name="molestiae quas omnis qui illo excepturi esse"
    onChange={[MockFunction]}
    readOnly={false}
    required={true}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
  />
  <p>
    molestias placeat exercitationem unde minus facere iste
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 11`] = `
<div>
  <textarea
    cols={30}
    disabled={true}
    id="sunt modi aut dolorem in non dolorem"
    onChange={[MockFunction]}
    placeholder="quis voluptatem quod tempore temporibus impedit dolores"
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value={[Function]}
  />
  <p>
    voluptatibus omnis fugit labore quia et beatae
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 12`] = `
<div
  className="suscipit delectus aliquid et dicta ipsam qui"
>
  <textarea
    cols={30}
    disabled={false}
    id="adipisci voluptatum id porro ea numquam consequatur"
    maxLength={128}
    name="et et voluptas repellendus nisi corrupti quas"
    onChange={[MockFunction]}
    placeholder="ut officiis sed illum et ea quia"
    readOnly={true}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="sunt consectetur perspiciatis nisi repellendus quia voluptate"
  />
  <p>
    aut laborum at non nulla sit officia
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 13`] = `
<div
  className="modi soluta hic omnis atque inventore vitae"
>
  <textarea
    cols={30}
    disabled={false}
    id="doloremque non sit a perferendis soluta dignissimos"
    name="cum non ea sed consequatur ut sapiente"
    onChange={[MockFunction]}
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="eos quia quam sit optio rerum quasi"
  />
  <p>
    in eos sunt ipsum saepe veritatis quisquam
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 14`] = `
<div
  className="animi quis unde animi aut sapiente fuga"
>
  <textarea
    cols={30}
    disabled={true}
    id="eius dolore quia error sed asperiores amet"
    name="et quos repellat distinctio libero mollitia facilis"
    onChange={[MockFunction]}
    placeholder="laboriosam tempora aut cupiditate aut non ipsam"
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="quia et optio praesentium iure dicta ab"
  />
  <p>
    enim est aspernatur voluptatem consequatur temporibus quasi
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 15`] = `
<div
  className="dolores eaque quis omnis ut praesentium vel"
>
  <textarea
    cols={30}
    disabled={false}
    id="numquam ut et quia cupiditate minus cum"
    maxLength={2042}
    name="est itaque qui dolorem voluptatem laboriosam iusto"
    onChange={[MockFunction]}
    placeholder="sint architecto corrupti quia nostrum molestiae quae"
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="quia blanditiis qui sed a dolores voluptatem"
  />
  <p>
    officia maiores cumque consequuntur rerum aspernatur nulla
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 16`] = `
<div>
  <textarea
    cols={30}
    disabled={false}
    id="aut omnis blanditiis accusamus ut odit quo"
    maxLength={114}
    name="enim facere illo nemo iusto iste illo"
    onChange={[MockFunction]}
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="sunt eligendi nihil porro suscipit sit eos"
  />
  <p>
    consequuntur qui repellat laudantium et qui et
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 17`] = `
<div
  className="possimus voluptas eaque expedita iste velit et"
>
  <textarea
    cols={30}
    disabled={false}
    id="aperiam quia est et suscipit ipsa et"
    maxLength={294}
    onChange={[MockFunction]}
    placeholder="voluptatem delectus aliquam et ipsam dolores voluptatem"
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="qui exercitationem eum laudantium autem et recusandae"
  />
  <p>
    suscipit reiciendis in exercitationem et error tempore
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 18`] = `
<div>
  <Label
    htmlFor="aspernatur soluta possimus voluptas sed esse rem"
  >
    fugiat rerum officiis assumenda nihil ad ut
  </Label>
  <textarea
    cols={30}
    disabled={false}
    id="aspernatur soluta possimus voluptas sed esse rem"
    maxLength={664}
    onChange={[MockFunction]}
    placeholder="id quisquam aspernatur quis consequatur eos perferendis"
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
  />
  <p>
    quidem excepturi temporibus facere ut velit odit
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 19`] = `
<div>
  <Label
    htmlFor="ratione saepe id officia pariatur et tenetur"
  >
    nobis sit nihil sint occaecati dolore autem
    <span
      style={
        Object {
          "color": "#B20000",
        }
      }
    >
      *
    </span>
  </Label>
  <textarea
    cols={30}
    disabled={false}
    id="ratione saepe id officia pariatur et tenetur"
    maxLength={1711}
    onChange={[MockFunction]}
    placeholder="voluptatibus minima consequatur eligendi excepturi fugit beatae"
    readOnly={false}
    required={true}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="reprehenderit ex nisi repudiandae qui ex unde"
  />
  <p>
    est ab quia ab sed voluptas ullam
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 20`] = `
<div
  className="qui est quis est veniam eum laudantium"
>
  <textarea
    cols={30}
    disabled={false}
    id="animi excepturi laborum veniam asperiores et consectetur"
    onChange={[MockFunction]}
    placeholder="sed enim provident ut qui odit nam"
    readOnly={true}
    required={true}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value={[Function]}
  />
  <p>
    ut dolor et placeat rerum sequi ex
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 21`] = `
<div
  className="officiis voluptatem veniam aut voluptas suscipit sit"
>
  <textarea
    cols={30}
    disabled={false}
    id="adipisci fugiat officiis enim voluptatem adipisci sequi"
    maxLength={2706}
    onChange={[MockFunction]}
    readOnly={true}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value={[Function]}
  />
  <p>
    nisi corporis nulla officia rerum incidunt porro
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 22`] = `
<div>
  <textarea
    cols={30}
    disabled={false}
    id="amet at asperiores libero officia totam corporis"
    name="rem sit eligendi ipsa molestiae aut doloribus"
    onChange={[MockFunction]}
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value={[Function]}
  />
  <p>
    aut nostrum repudiandae harum neque ut dolores
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 23`] = `
<div>
  <textarea
    cols={30}
    disabled={false}
    id="qui qui architecto aliquid iure eos expedita"
    onChange={[MockFunction]}
    placeholder="blanditiis iste animi ipsa est molestias deserunt"
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="quos eos iste aut inventore optio tempora"
  />
  <p>
    iste ut a ut exercitationem voluptatem nam
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 24`] = `
<div
  className="qui necessitatibus quae tempora voluptatem autem voluptas"
>
  <textarea
    cols={30}
    disabled={false}
    id="in quod et ea quasi perferendis voluptatem"
    maxLength={2696}
    name="nemo consequatur voluptatum eligendi eos magni iste"
    onChange={[MockFunction]}
    placeholder="et fuga adipisci saepe quam quia veritatis"
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="aut est molestias omnis nisi vel omnis"
  />
  <p>
    numquam et culpa nulla ipsum et explicabo
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 25`] = `
<div>
  <textarea
    cols={30}
    disabled={false}
    id="vel quaerat fuga enim numquam esse quia"
    name="aperiam libero ipsa voluptates quaerat recusandae accusamus"
    onChange={[MockFunction]}
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
  />
  <p>
    sunt aut quam provident voluptates perspiciatis nulla
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 26`] = `
<div
  className="aspernatur earum ex officia veritatis eos aut"
>
  <textarea
    cols={30}
    disabled={false}
    id="maiores pariatur amet harum maxime a repellendus"
    onChange={[MockFunction]}
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="quia eum temporibus praesentium ducimus facere esse"
  />
  <p>
    dolor delectus unde reprehenderit deleniti nihil ex
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 27`] = `
<div>
  <textarea
    cols={30}
    disabled={true}
    id="maiores nihil quisquam id omnis sint voluptatum"
    maxLength={3462}
    onChange={[MockFunction]}
    placeholder="assumenda odio id id repellat maiores iure"
    readOnly={true}
    required={true}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="quod rerum suscipit impedit veniam perferendis est"
  />
  <p>
    similique optio hic perspiciatis dolor quod doloribus
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 28`] = `
<div
  className="velit velit perspiciatis mollitia ut corrupti sunt"
>
  <textarea
    cols={30}
    disabled={false}
    id="culpa possimus consectetur voluptas perferendis dicta iste"
    onChange={[MockFunction]}
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="saepe dicta qui asperiores nihil illum ut"
  />
  <p>
    voluptatem rerum et cumque quo sed aut
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 29`] = `
<div>
  <textarea
    cols={30}
    disabled={false}
    id="dolore labore vitae aperiam voluptas impedit cumque"
    maxLength={3472}
    onChange={[MockFunction]}
    readOnly={false}
    required={true}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="aut eaque dignissimos incidunt quis non ut"
  />
  <p>
    animi et sapiente qui minima ea beatae
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 30`] = `
<div>
  <textarea
    cols={30}
    disabled={false}
    id="consequatur vero quis voluptates voluptates similique laboriosam"
    maxLength={1411}
    name="molestiae modi expedita quaerat eos aut itaque"
    onChange={[MockFunction]}
    placeholder="quis incidunt provident assumenda sit pariatur quam"
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value={[Function]}
  />
  <p>
    molestias odio qui sit et voluptate vitae
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 31`] = `
<div>
  <textarea
    cols={30}
    disabled={true}
    id="quidem aliquam et quaerat expedita sunt est"
    maxLength={4666}
    onChange={[MockFunction]}
    readOnly={true}
    required={true}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value={[Function]}
  />
  <p>
    eaque adipisci eos ab error velit modi
  </p>
</div>
`;

exports[`<TextArea /> matches snapshot 32`] = `
<div
  className="sint enim ratione cupiditate cumque suscipit fuga"
>
  <textarea
    cols={30}
    disabled={false}
    id="sit sunt exercitationem aliquam unde quisquam vel"
    maxLength={4071}
    name="saepe et qui aut enim culpa eius"
    onChange={[MockFunction]}
    readOnly={false}
    required={false}
    rows={10}
    style={
      Object {
        "marginBottom": "0px",
      }
    }
    value="molestiae temporibus non dolorum natus reprehenderit facere"
  />
  <p>
    expedita veritatis pariatur magni ut explicabo iste
  </p>
</div>
`;
